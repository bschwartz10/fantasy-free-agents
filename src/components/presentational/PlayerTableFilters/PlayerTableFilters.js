import React from 'react';
import { FormGroup, Checkbox, FormControl } from 'react-bootstrap';

import './PlayerTableFilters.css';

const PlayerTableFilters = ({toggleChange, handleChange}) => {

  return (
    <div className="PlayerTableFilters">
      <FormGroup>
        <Checkbox className="qb" inline onChange={(e) => toggleChange(e, 'QB')}>QB</Checkbox>
        <Checkbox className="rb" inline onChange={(e) => toggleChange(e, 'RB')}>RB</Checkbox>
        <Checkbox className="wr" inline onChange={(e) => toggleChange(e, 'WR')}>WR</Checkbox>
        <Checkbox className="te" inline onChange={(e) => toggleChange(e, 'TE')}>TE</Checkbox>
      </FormGroup>
      <FormControl type="text" placeholder="Search by Name" onChange={(e) => handleChange(e)} />
    </div>
  )
}

export default PlayerTableFilters;
