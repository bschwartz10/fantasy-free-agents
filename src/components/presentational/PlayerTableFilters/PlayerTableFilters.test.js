import React from 'react'
import { shallow } from 'enzyme'
import PlayerTableFilters from './PlayerTableFilters'

describe('PlayerTableFilters component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PlayerTableFilters toggleChange={jest.fn()} handleClick={jest.fn()} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('render boxes with correct labels', () => {
    expect(wrapper.find('.qb').first().children().text()).toEqual('QB')
    expect(wrapper.find('.rb').first().children().text()).toEqual('RB')
    expect(wrapper.find('.wr').first().children().text()).toEqual('WR')
    expect(wrapper.find('.te').first().children().text()).toEqual('TE')
  })

  it('should match snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

})
