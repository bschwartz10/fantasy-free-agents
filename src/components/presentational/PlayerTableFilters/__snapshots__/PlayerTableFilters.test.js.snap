// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerTableFilters component should match snapshot with all data passed in correctly 1`] = `
<div
  className="PlayerTableFilters"
>
  <FormGroup
    bsClass="form-group"
  >
    <Checkbox
      bsClass="checkbox"
      className="qb"
      disabled={false}
      inline={true}
      onChange={[Function]}
      title=""
    >
      QB
    </Checkbox>
    <Checkbox
      bsClass="checkbox"
      className="rb"
      disabled={false}
      inline={true}
      onChange={[Function]}
      title=""
    >
      RB
    </Checkbox>
    <Checkbox
      bsClass="checkbox"
      className="wr"
      disabled={false}
      inline={true}
      onChange={[Function]}
      title=""
    >
      WR
    </Checkbox>
    <Checkbox
      bsClass="checkbox"
      className="te"
      disabled={false}
      inline={true}
      onChange={[Function]}
      title=""
    >
      TE
    </Checkbox>
  </FormGroup>
  <FormControl
    bsClass="form-control"
    componentClass="input"
    onChange={[Function]}
    placeholder="Search by Name"
    type="text"
  />
</div>
`;
